src/ip_notifier.cpp

```diff
@@ -80,6 +82,10 @@ namespace libtorrent
 		{
 			m_hnd.async_wait([this,cb](error_code const& ec) { on_notify(ec, 0, cb); });
 		}
+		else
+		{
+			cb(error_code(err, system_category()));
```

Comment:
I think, technically, you need to ``post()`` the call to the callback here. To make sure it always run after the current handler (which may not expect to be recursively invoked)